napari/utils/events/evented_model.py

```diff
@@ -104,6 +104,15 @@ def __new__(mcs, name, bases, namespace, **kwargs):
         for name, attr in namespace.items():
             if isinstance(attr, property):
                 cls.__properties__[name] = attr
+                # determine compare operator
+                if (
+                    hasattr(attr.fget, "__annotations__")
+                    and "return" in attr.fget.__annotations__
+                ):
+                    cls.__eq_operators__[name] = pick_equality_operator(
+                        attr.fget.__annotations__["return"]
+                    )
+
```

Comment:
Did we have any guarantee/test/benchmarks for pick_equality_operator?

Maybe put compression in `try ... except ... ` block?

Reply:
It is dict lookup with traverse through `__mro__`. Currently it will return `operator.eq` for `BasicModel`. 

I'm not against reversing this change. I try to understand problematic scenarios. 
Maybe we should also check type equality before call eq operator?

Reply:
But we have some types in napari (like the `Encoding`s introduced by @andy-sweet) that can take the same field, give the same "output" but be completely different object. I had issues with them before due to this switcheroo mechanism :P

Reply:
Need to catch up properly, but to clarify the encodings are protocols. If I understand the code here correctly (and I'm not sure I do), but I think if `pick_equality_operator` works for the field type, it should also work for the property getter return type.

Reply:
The problem may occur when the property return a different type than is annotated.